.github/workflows/container.yaml

```diff
@@ -57,8 +59,48 @@ jobs:
             "type=ref,event=tag"
             "type=ref,event=pr"
 
-      - name: Build and push Docker image
-        uses: docker/build-push-action@v6
+      - name: Build Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: Dockerfile.${{ matrix.dockerfile }}
+          load: true
+          push: false
+          tags: ${{ env.TEST_TAG }}
+          labels: ${{ steps.meta.outputs.labels }}
+          build-args: |
+            VARIANT=${{ matrix.variant }}
+
+      - name: Checkout fakemachine (for unit tests)
```

Comment:
did try to checkout latest tag; but it was too much boilerplate code. main should already be tested before merging so it should be fine.